tests/k8s-azure/manifest/cluster-api/vmss-multi-nodepool.yaml

```diff
@@ -454,73 +454,131 @@ data:
-          hostPath:
-            path: /etc/kubernetes
-        - name: etc-ssl
-          hostPath:
-            path: /etc/ssl
-        - name: msi
-          hostPath:
-            path: /var/lib/waagent/ManagedIdentity-Settings
+      replicas: 1
+      selector:
+        matchLabels:
+          tier: control-plane
+          component: cloud-controller-manager
+    template:
+      metadata:
+        labels:
+          tier: control-plane
+          component: cloud-controller-manager
+      spec:
+        priorityClassName: system-node-critical
+        hostNetwork: true
+        nodeSelector:
+          node-role.kubernetes.io/master: ""
+        serviceAccountName: cloud-controller-manager
+        tolerations:
+          - key: node-role.kubernetes.io/master
+            effect: NoSchedule
+        containers:
+          - name: cloud-controller-manager
+            image: ${AZURE_CLOUD_CONTROLLER_MANAGER_IMG}
+            imagePullPolicy: IfNotPresent
+            command: ["cloud-controller-manager"]
+            args:
+              - "--allocate-node-cidrs=true"
+              - "--enable-dynamic-reloading=true"
+              - "--cloud-config=/etc/kubernetes/cloud-config/azure.json"
+              - "--cloud-provider=azure"
+              - "--cluster-cidr=10.244.0.0/16"
+              - "--cluster-name=${CLUSTER_NAME}"
+              - "--controllers=*,-cloud-node" # disable cloud-node controller
+              - "--configure-cloud-routes=true" # "false" for Azure CNI and "true" for other network plugins
+              - "--leader-elect=true"
+              - "--route-reconciliation-period=10s"
+              - "--v=4"
+              - "--port=10267"
+            resources:
+              requests:
+                cpu: 100m
+                memory: 128Mi
+              limits:
+                cpu: "4"
+                memory: 2Gi
+            livenessProbe:
+              httpGet:
+                path: /healthz
+                port: 10267
+              initialDelaySeconds: 20
+              periodSeconds: 10
+              timeoutSeconds: 5
+            volumeMounts:
+              - name: etc-kubernetes
+                mountPath: /etc/kubernetes
+              - name: cloud-config
+                mountPath: /etc/kubernetes/cloud-config
+              - name: etc-ssl
+                mountPath: /etc/ssl
+                readOnly: true
+              - name: msi
+                mountPath: /var/lib/waagent/ManagedIdentity-Settings
+                readOnly: true
+        volumes:
+          - name: etc-kubernetes
+            hostPath:
+              path: /etc/kubernetes
+          - name: cloud-config
+            secret:
+              secretName: azure-cloud-config
+          - name: etc-ssl
+            hostPath:
+              path: /etc/ssl
+          - name: msi
+            hostPath:
+              path: /var/lib/waagent/ManagedIdentity-Settings
+    ---
+    apiVersion: v1
+    kind: Secret
```

Comment:
I think capz should already create a secret, so we don't need to add it here?  @nilo19 do you know the secret name, or host path should be used instead just for capz?

Reply:
We don't need a secret here in this template, and host path either. The capz will create a default one and the volume is defined in L533.

Reply:
Discussed offline. Since this yaml file is for capz, then keep it as it was.